### PR TITLE
fix dogu sort

### PR DESCRIPTION
**Pull request by meiserloh**

### CODE REVIEW

**Comment by robertauer** [comment]
Please update the changelog

File: core/dogu-sort.go [hunk 1]
```diff
@@ -64,48 +65,47 @@ func contains(slice []Dependency, item string) bool {
 	return false
 }
 
-func (bd *sortByDependency) Len() int {
```
Comment (nhinze23) [inline core/dogu-sort.go:67]: Why did you delete these public functions? This results in a breaking change.

Comment (nhinze23) [reply core/dogu-sort.go:67]: Ah we only needed this for the insertion sort

File: core/dogu-sort.go [hunk 2]
```diff
@@ -19,7 +20,7 @@ func SortDogusByInvertedDependency(dogus []*Dogu) []*Dogu {
 
 	orderedAsc := []*Dogu{}
 	for i := len(orderedDesc) - 1; i >= 0; i-- {
-		orderedAsc = append(orderedAsc, dogus[i])
+		orderedAsc = append(orderedAsc, orderedDesc[i])
```
[the remark this answers is not in the document]
Comment (nhinze23) [reply core/dogu-sort.go:23]: I think we could use `ToposortR` here

File: resources/test/dogu-sort-004.json [hunk 3]
```diff
@@ -0,0 +1,6609 @@
+[
```
Comment (nhinze23) [inline resources/test/dogu-sort-004.json:1]: Nice testdata but i think only using necessary attributes makes it easier to perceive the dogus to test.

File: core/dogu-sort.go [hunk 4]
```diff
@@ -64,48 +65,47 @@ func contains(slice []Dependency, item string) bool {
...
-	leftDependenciesRecursive := bd.getAllDoguDependenciesRecursive(bd.dogus[i])
-	rightDependenciesRecursive := bd.getAllDoguDependenciesRecursive(bd.dogus[j])
-
-	leftDependenciesDirect := bd.dogus[i].GetAllDependenciesOfType(DependencyTypeDogu)
-	rightDependenciesDirect := bd.dogus[j].GetAllDependenciesOfType(DependencyTypeDogu)
+func (bd *sortByDependency) sortDogus() []*Dogu {
+	dependencyEdges := bd.getDependencyEdges()
+	sorted, err := toposort.Toposort(dependencyEdges)
+	if err != nil {
+		return nil
```
Comment (meiserloh) [inline core/dogu-sort.go:72]: Write error log in error cases

**Comment by nhinze23** [comment]
lgtm
